pkgs/build-support/fetchpypilegacy/default.nix

```diff
@@ -18,28 +33,31 @@
   name ? null,
 }:
 let
-  urls' = urls ++ lib.optional (url != null) url;
+  urls' = urls ++ optional (url != null) url;
 
-  pathParts = lib.filter ({ prefix, path }: "NETRC" == prefix) builtins.nixPath;
-  netrc_file =
-    if (pathParts != [ ])
-    then (lib.head pathParts).path
-    else "";
+  pathParts = filter ({ prefix, path }: "NETRC" == prefix) builtins.nixPath;
+  netrc_file = if (pathParts != [ ]) then (head pathParts).path else "";
 
 in
 # Assert that we have at least one URL
-assert urls' != [ ]; runCommand file
-  ({
-    nativeBuildInputs = [ python3 ];
-    impureEnvVars = lib.fetchers.proxyImpureEnvVars;
-    outputHashMode = "flat";
-    # if hash is empty select a default algo to let nix propose the actual hash.
-    outputHashAlgo = if hash == "" then "sha256" else null;
-    outputHash = hash;
-    NETRC = netrc_file;
-  }
-  // (lib.optionalAttrs (name != null) {inherit name;}))
+assert urls' != [ ];
+runCommand file
+  (
+    {
+      nativeBuildInputs = [ python3 ];
+      inherit impureEnvVars;
+      outputHashMode = "flat";
+      # if hash is empty select a default algo to let nix propose the actual hash.
+      outputHashAlgo = if hash == "" then "sha256" else null;
+      outputHash = hash;
+      NETRC = netrc_file;
+    }
+    // optionalAttrs (name != null) { inherit name; }
+    // optionalAttrs (!inPureEvalMode) { NETRC = netrc_file; }
```

Comment:
This is superfluous?

Reply:
That was a bug that was already present in the original commit https://github.com/nix-community/poetry2nix/commit/f625ae2e51ec04a648e8108700f63864e0029db4.

We shouldn't add `NETRC` when in pure eval because it originates from `builtins.nixPath` which isn't available in pure eval.

Reply:
Hey, my review found something! I'm counting that as a win. 😁